New Script: Zitadel

dave-yap opened the pull request:
## ✍️ Description
Zitadel is an open-source identity and access management (IAM) solution designed to provide secure authentication, authorization, and user management for modern applications and services. Built with a focus on flexibility, scalability, and security, Zitadel offers a comprehensive set of features for developers and organizations looking to implement robust identity management.

- - -
- Related Issue: #
- Related PR: #
- Related Discussion: #
- - - 


## ✅ Prerequisites
The following steps must be completed for the pull request to be considered:  
- [✅] Self-review performed (I have reviewed my code to ensure it follows established patterns and conventions.)  
- [✅] Testing performed (I have thoroughly tested my changes and verified expected functionality.)

## 🛠️ Type of Change
Please check the relevant options:  
- [] Bug fix (non-breaking change that resolves an issue)  
- [] New feature (non-breaking change that adds functionality)  
- [] Breaking change (fix or feature that would cause existing functionality to change unexpectedly)  
- [✅] New script (a fully functional and thoroughly tested script or set of scripts)  

---
## 📋 Additional Information (optional)
Provide any extra context or screenshots about the feature or fix here.  

This script exposes port 8080 with TLS disabled but users are able to easily change the config.yaml to include TLS solutions. The service needed to be ran twice in the install-script as initially the console is only accessible via localhost. As it is a LXC script, I do not see a way to access the console externally. Thus, the setup was ran again in order to have a nice admin username + access externally. Any changes done to the config.yaml file required an execution of the included bash script to ensure the database is migrated properly. This is based on Zitadel's own documentations.

dave-yap committed:
Create zitadel-install.sh

dave-yap committed:
Create zitadel.json

dave-yap committed:
Create zitadel.sh

dave-yap committed:
Update zitadel.sh

Edit reference back to upstream build.func

tremor021 commented on json/zitadel.json:
```diff
@@ -0,0 +1,43 @@
+    "updateable": true,
+    "privileged": false,
+    "interface_port": 8080,
+    "documentation": null,
```
```suggestion
    "documentation": "https://zitadel.com/docs/guides/overview",
```

dave-yap committed:
Update zitadel.json

Co-authored-by: Slaviša Arežina <58952836+tremor021@users.noreply.github.com>

tremor021 commented on ct/zitadel.sh:
```diff
@@ -0,0 +1,70 @@
+        i686) ARCH="386";;
+        i386) ARCH="386";;
+    esac
+    if [[ "${LATEST}" != "$(cat /opt/zitadel_version.txt | grep -oP '\d+\.\d+\.\d+')" ]] || [[ ! -f /opt/zitadel_version.txt ]]; then
```
```suggestion
    if [[ "${RELEASE}" != "$(cat /opt/zitadel_version.txt | grep -oP '\d+\.\d+\.\d+')" ]] || [[ ! -f /opt/zitadel_version.txt ]]; then
```

tremor021 commented on ct/zitadel.sh:
```diff
@@ -0,0 +1,70 @@
+        zitadel setup --masterkeyFile /opt/zitadel/.masterkey --config /opt/zitadel/config.yaml --init-projections=true &>/dev/null
+        systemctl start zitadel.service
+        echo "v${LATEST}" > /opt/zitadel_version.txt
+        msg_ok "Updated ${APP} to v${LATEST}"
```
```suggestion
        msg_ok "Updated ${APP} to v${RELEASE}"
```

tremor021 commented on ct/zitadel.sh:
```diff
@@ -0,0 +1,70 @@
+        echo "v${LATEST}" > /opt/zitadel_version.txt
+        msg_ok "Updated ${APP} to v${LATEST}"
+    else
+        msg_ok "No update required. ${APP} is already at v${LATEST}"
```
```suggestion
        msg_ok "No update required. ${APP} is already at v${RELEASE}"
```

dave-yap committed:
Update zitadel.sh

Co-authored-by: Slaviša Arežina <58952836+tremor021@users.noreply.github.com>

dave-yap committed:
Update zitadel.sh

Co-authored-by: Slaviša Arežina <58952836+tremor021@users.noreply.github.com>

dave-yap committed:
Update zitadel.sh

Co-authored-by: Slaviša Arežina <58952836+tremor021@users.noreply.github.com>

dave-yap committed:
Update zitadel.sh

Co-authored-by: Slaviša Arežina <58952836+tremor021@users.noreply.github.com>

bvdberg01 commented on ct/zitadel.sh:
```diff
@@ -0,0 +1,70 @@
+#!/usr/bin/env bash
+source <(curl -s https://raw.githubusercontent.com/community-scripts/ProxmoxVE/main/misc/build.func)
+# Copyright (c) 2021-2025 tteck
```
```suggestion
# Copyright (c) 2021-2025 community-scripts ORG
```

dave-yap committed:
Update zitadel.sh

tremor021 commented on install/zitadel-install.sh:
```diff
@@ -0,0 +1,168 @@
+msg_ok "Installed PostgreSQL"
+
+msg_info "Installing Zitadel"
+LATEST=$(curl -si https://github.com/zitadel/zitadel/releases/latest | grep location: | cut -d '/' -f 8 | tr -d '\r') &>/dev/null
```
```suggestion
RELEASE=$(curl -si https://github.com/zitadel/zitadel/releases/latest | grep location: | cut -d '/' -f 8 | tr -d '\r') &>/dev/null
```
Keep it as RELEASE, its like that in all scripts

tremor021 commented on install/zitadel-install.sh:
```diff
@@ -0,0 +1,168 @@
+    i686) ARCH="386";;
+    i386) ARCH="386";;
+esac
+wget -q -c https://github.com/zitadel/zitadel/releases/download/$LATEST/zitadel-linux-$ARCH.tar.gz -O - | tar -xz &>/dev/null
```
```suggestion
wget -q -c https://github.com/zitadel/zitadel/releases/download/$RELEASE/zitadel-linux-$ARCH.tar.gz -O - | tar -xz &>/dev/null
```

dave-yap committed:
Update zitadel-install.sh

Co-authored-by: Slaviša Arežina <58952836+tremor021@users.noreply.github.com>

dave-yap committed:
Update zitadel-install.sh

Co-authored-by: Slaviša Arežina <58952836+tremor021@users.noreply.github.com>

tremor021 commented on ct/zitadel.sh:
```diff
@@ -0,0 +1,70 @@
+        rm -rf zitadel-linux-$ARCH
+        zitadel setup --masterkeyFile /opt/zitadel/.masterkey --config /opt/zitadel/config.yaml --init-projections=true &>/dev/null
+        systemctl start zitadel.service
+        echo "v${LATEST}" > /opt/zitadel_version.txt
```
```suggestion
        echo "v${RELEASE}" > /opt/zitadel_version.txt
```

tremor021 reviewed (changes requested):
Small things

bvdberg01 commented on install/zitadel-install.sh:
```diff
@@ -0,0 +1,168 @@
+  postgres:
+    Host: localhost
+    Port: 5432
+    Database: zitadel
```
Use variable DB_NAME

dave-yap replied:
Possible even within EOF parts? New to scripting, would love an insight!

michelroegl-brunner replied:
use ${VariableName}

dave-yap replied:
The edit has been done, please re-review thanks! The other variables are hardcoded for the user to change as per their needs; this spins up a test instance with everything defaulted. It can be ran without the variables too, so should I remove the variables instead?

dave-yap commented:
> Small things

No worries, thanks for letting me know. Was following the Zitadel docs and didn't realize I scripted around it instead of following the community guide. My bad!

bvdberg01 commented on install/zitadel-install.sh:
```diff
@@ -0,0 +1,168 @@
+sudo /usr/share/postgresql-common/pgdg/apt.postgresql.org.sh <<< return &>/dev/null
+DB_NAME="zitadel"
+DB_USER="zitadel"
+DB_PASS="zitadel"
```
Generate password like this
```suggestion
DB_PASS=$(openssl rand -base64 18 | tr -dc 'a-zA-Z0-9' | cut -c1-13)
```

dave-yap replied:
Not sure how to implement this into the config.yaml but will have a look!

bvdberg01 replied:
> Not sure how to implement this into the config.yaml but will have a look!

Here an example how I did it in baikal script
```
cat <<EOF >/opt/baikal/config/baikal.yaml
database:
    backend: pgsql
    pgsql_host: localhost
    pgsql_dbname: $DB_NAME
    pgsql_username: $DB_USER
    pgsql_password: $DB_PASS
EOF
```

dave-yap replied:
Looks good, implementing now!

dave-yap committed:
Update zitadel.sh

Co-authored-by: bvdberg01 <74251551+bvdberg01@users.noreply.github.com>

bvdberg01 commented on json/zitadel.json:
```diff
@@ -0,0 +1,43 @@
+    "name": "Zitadel",
+    "slug": "Zitadel",
+    "categories": [
+        0
```
I think catergory Authentication & Security fits better then Miscellaneous
```suggestion
        6
```

dave-yap committed:
Update zitadel.json

Co-authored-by: bvdberg01 <74251551+bvdberg01@users.noreply.github.com>

dave-yap committed:
Use declared variables in config files

dave-yap committed:
Merge branch 'community-scripts:main' into main

michelroegl-brunner commented:
Are the arch differenzes necessery? Better to just use the linux_amd64 version. There is a different Project on github for Proxmox on ARM. https://github.com/asylumexp/Proxmox
Mybee he will also accept a PR with the ARM stuff.

MickLesk commented:
Yes, I would say that we don't really need all the Arch comparisons. One amd64 variant is enough. The other one is done by @asylumexp in the other repo (arm64)

dave-yap committed:
Remove other architectures

dave-yap commented:
Does the other repo use the same flow? Can I simply push a arm64 variant over to the other repo?

asylumexp commented:
> Does the other repo use the same flow? Can I simply push a arm64 variant over to the other repo?

No, you just push to the community-scripts repo, and I just edit the scripts so they run on arm64.

michelroegl-brunner commented on install/zitadel-install.sh:
```diff
@@ -0,0 +1,158 @@
+    ca-certificates \
+    wget \
+    sed \
+    sudo
```
mc is missing

michelroegl-brunner commented on install/zitadel-install.sh:
```diff
@@ -0,0 +1,158 @@
+
+msg_info "Installing Zitadel"
+RELEASE=$(curl -si https://github.com/zitadel/zitadel/releases/latest | grep location: | cut -d '/' -f 8 | tr -d '\r') &>/dev/null
+ARCH=amd64
```
Remove the variable and put it directly in the link

michelroegl-brunner commented on install/zitadel-install.sh:
```diff
@@ -0,0 +1,158 @@
+ARCH=amd64
+wget -q -c https://github.com/zitadel/zitadel/releases/download/$RELEASE/zitadel-linux-$ARCH.tar.gz -O - | tar -xz &>/dev/null
+mv zitadel-linux-$ARCH/zitadel /usr/local/bin
+echo -e "$(zitadel -v | grep -oP '\d+\.\d+\.\d+')" > /opt/zitadel_version.txt 
```
Bust use the release variable

dave-yap replied:
Sorry, but isn't the release variable being used here?

michelroegl-brunner commented on json/zitadel.json:
```diff
@@ -0,0 +1,43 @@
+            "resources": {
+                "cpu": 2,
+                "ram": 2048,
+                "hdd": 12,
```
eather here or in zitadell.sh the values are wrong

michelroegl-brunner commented on json/zitadel.json:
```diff
@@ -0,0 +1,43 @@
+    "notes": [
+        {
+            "text": "Application credentials: `cat ~/zitadel.creds`",
+            "type": "info"
```
are those the same as the credentials above? if so you can remove this here.

dave-yap replied:
Nope they are different, zitadel.creds store the credentials of the used variables in the config.yaml file in the initial setup.

dave-yap committed:
Update to fit changes requested

Include mc for install; removal of variable ARCH and put into direct links; correct the default resources required

dave-yap committed:
Merge branch 'community-scripts:main' into main

MickLesk commented on install/zitadel-install.sh:
```diff
@@ -0,0 +1,158 @@
+msg_info "Installing Dependencies (Patience)"
+$STD apt-get install -y \
+    curl \
+    sudo\
```
Space Missing

MickLesk commented on install/zitadel-install.sh:
```diff
@@ -0,0 +1,158 @@
+    mc \
+    ca-certificates \
+    wget \
+    sed
```
Not needed its in mc

dave-yap replied:
so ca-certificates, wget and sed are not needed yes? I'll remove those

MickLesk replied:
I mean sed. 
Ca idk

MickLesk commented on install/zitadel-install.sh:
```diff
@@ -0,0 +1,158 @@
+    echo "DB_ADMIN_USER: $DB_ADMIN_USER"
+    echo "DB_ADMIN_PASS: $DB_ADMIN_PASS"
+} >> ~/zitadel.creds
+systemctl enable -q --now postgresql
```
Not needed

MickLesk commented on install/zitadel-install.sh:
```diff
@@ -0,0 +1,158 @@
+    echo "DB_ADMIN_PASS: $DB_ADMIN_PASS"
+} >> ~/zitadel.creds
+systemctl enable -q --now postgresql
+sudo -u postgres psql -c "CREATE USER $DB_USER WITH PASSWORD '$DB_PASS';" &>/dev/null
```
Add STD before instead >\dev\null

MickLesk commented on install/zitadel-install.sh:
```diff
@@ -0,0 +1,158 @@
+sudo -u postgres psql -c "CREATE USER $DB_USER WITH PASSWORD '$DB_PASS';" &>/dev/null
+sudo -u postgres psql -c "CREATE USER $DB_ADMIN_USER WITH PASSWORD '$DB_ADMIN_PASS' SUPERUSER;" &>/dev/null
+sudo -u postgres psql -c "CREATE DATABASE $DB_NAME OWNER $DB_ADMIN_USER;" &>/dev/null
+systemctl restart -q postgresql
```
Not needed

MickLesk commented on install/zitadel-install.sh:
```diff
@@ -0,0 +1,158 @@
+customize
+
+msg_info "Cleaning up"
+rm -rf zitadel-linux-amd64
```
Add path

dave-yap committed:
Update zitadel.sh

Co-authored-by: bvdberg01 <74251551+bvdberg01@users.noreply.github.com>

dave-yap committed:
Update zitadel-install.sh

Co-authored-by: bvdberg01 <74251551+bvdberg01@users.noreply.github.com>

dave-yap committed:
Update zitadel-install.sh

Co-authored-by: bvdberg01 <74251551+bvdberg01@users.noreply.github.com>

dave-yap commented on install/zitadel-install.sh:
```diff
@@ -0,0 +1,158 @@
+RELEASE=$(curl -si https://github.com/zitadel/zitadel/releases/latest | grep location: | cut -d '/' -f 8 | tr -d '\r') &>/dev/null
+wget -q -c https://github.com/zitadel/zitadel/releases/download/$RELEASE/zitadel-linux-amd64.tar.gz -O - | tar -xz &>/dev/null
+mv zitadel-linux-amd64/zitadel /usr/local/bin
+echo -e "$(zitadel -v | grep -oP '\d+\.\d+\.\d+')" > /opt/zitadel_version.txt 
```
/opt/zitadel_version.txt is a reference file for the current installed version. RELEASE variable is from the parsed value.

dave-yap committed:
Made changes to fit suggestions

dave-yap committed:
Update zitadel-install.sh

correct version output

dave-yap committed:
Update zitadel-install.sh

dave-yap committed:
Merge branch 'main' of https://github.com/dave-yap/ProxmoxVE

dave-yap committed:
Update path for version.txt

dave-yap committed:
Merge branch 'community-scripts:main' into main

dave-yap commented:
Should all be fixed now

MickLesk committed:
Set update part default to our project defaults

dave-yap commented:
There'll be a recurring `v` here at this line:  `msg_info "Updating $APP to v${RELEASE}"`, no need for the `v` before calling the RELEASE variable

michelroegl-brunner committed:
Update zitadel.sh, Remove v befor ${RELEASE}

michelroegl-brunner committed:
Update zitadel-install.sh

michelroegl-brunner commented:
Removed the v before ${RELEASE} and added the Version to File part in zitadel-install.sh
